### DISCUSSION TITLE
Upcoming target rates

### DISCUSSION THREAD

**Discussion by Dave-Parry** [BottlecapDave/HomeAssistant-OctopusEnergy#528] [Ideas]
Is it possible to display which upcoming agile tariff rates are set to be on?
I have the upcoming rates displayed but would like to know which ones will be used preferably for the next day if after 16:00.

**Comment by BottlecapDave** [comment]
Hi. When you say "set to be on", what do you mean exactly?

You can use target rate sensors to find the lowest agile rates so you can turn things on/off which also expose the discovered lowest rates as an attribute on the sensor. This attribute will update once the previous set of rates are all in the past.

Or if you're just wanting a visual way of seeing upcoming rates, you can use one of the many solutions from the [community section](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/community.md)

**Comment by Dave-Parry** [comment]
Thanks for the reply David.
I already use those but I’m looking for the times when the octopus energy sensor turns on.
I’m on the agile tariff so I don’t have the planned despatching and other sensors the intelligent tariff seems to have.

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
The two things mentioned above are not tied to the intelligent tariff and the community content is geared more towards agile. When you say I'm looking for the times when the octopus energy tariff turns on, which sensor are you referring to?

**Comment by Dave-Parry** [comment]
Hi David.
Sorry , I’ve only just started with Home assistant so not completely up to speed with some of the terminology.
The sensor I mean’t was the one created when I configured the octopus energy add-on.
In my case I named it “binary_sensor.octopus_energy_target_my_charge_slots”.
What I’m looking to do is display the slots that change the sensor from off to on.

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
No problem, everyone has to start somewhere. Based on this [comment](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/500#discussioncomment-7562098), you'll want the specified template, but with the following adjustments

```
{% if state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') | length > 0 %}
{% for target_time in state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') -%}
⏰ {{ (target_time.valid_from | as_local).strftime('%H:%M') }} to {{ (target_time.valid_to | as_local).strftime('%H:%M') }}
{{ "\n" }}
{%- endfor %}
{%- else -%}
⏰ None
{%- endif %}
```

**Comment by Dave-Parry** [comment]
David,
Sorry to bother you again. I have been quite busy and only just got round to trying this.
I have amended the template you mentioned and I think I’m supposed to paste this into my configuration.yaml file.
However when I do I get a error message 
￼
Have tried adding a comma but doesn’t correct it.
Can you advise how to sort it out?
or have I got it completely wrong?

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
Where are you trying to display it? If you're wanting it on the dashboard, then it'll be the content of a markdown card.

**Comment by Dave-Parry** [comment]
Thanks for reply, 
I’m wanting to display the upcoming on periods.
So do I put the code in a markdown card, not in the configuration.yaml file?

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
If you're wanting to display it on your dashboard, then yeah the code goes into a markdown card with the dashboard

**Comment by Dave-Parry** [comment]
Dave,
Been trying to get this to work without any success.
I’ve inserted the following code into a markdown card but get a message   UndefinedError: dict object has no attribute ‘valid_from’

title: Octopus Intelligent Tariff
type: entities
show_header_toggle: false
entities:
  - entity: switch.octopus_smart_charging
    name: Smart charging enabled
  - entity: select.octopus_target_ready_by_time
    name: Ready by
  - entity: select.octopus_target_state_of_charge
    name: Charge to
  - type: custom:template-entity-row
    name: Planned slots
    secondary: >-
{% if state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') | length > 0 %}
{% for target_time in state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') -%}
⏰ {{ (target_time.valid_from | as_local).strftime('%H:%M') }} to {{ (target_time.valid_to | as_local).strftime('%H:%M') }}
{{ "\n" }}
{%- endfor %}
{%- else -%}
⏰ None
{%- endif %}

I also tried installing https://github.com/thomasloven/lovelace-template-entity-row  which was suggested a couple of posts below the one you quoted.
Still get the same message.
Have tried installing octopus intelligent integration, but I cant configure it as I am not on intelligent tariff. Consequently I don’t have the enteties
switch.octopus_smart_charging
select.octopus_target_ready_by_time
select.octopus_target_state_of_charge
Do you think it is possible to get round these problems or I is it a lost cause?
Thanks for your patience.

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
Hello. Since your initial post, theres been a major update to the integration with includes some changes to naming of some of the attributes. In your case you'll need to update valid_from to start and valid_to to end

As for your missing entities, those are not provided by this integration if you're not on an intelligent tariff so won't be applicable in your case.

**Comment by Dave-Parry** [comment]
Thank you very much Dave,
Thats sorted it out.

Regards
David Parry
***@***.***
